### PR TITLE
TDL-14567: Check linkedin ads accounts in discover mode

### PR DESCRIPTION
**Pull request by savan-chovatiya**
# Description of change
TDL-14567:The account_number should be validated in the discovery mode
- Added Account users API call for every account provided in the config for discover mode and raised an error if accounts are an invalid number or invalid for LinkedIn. 

# Manual QA steps
 - Verified that exception is raised with list of invalid accounts in discover mode if any accounts are invalid number or invalid for LinkedIn.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_linkedin_ads/__init__.py [hunk 1]
```diff
@@ -18,10 +18,11 @@
     'accounts'
 ]
 
-def do_discover(config):
+def do_discover(client, config):
 
     LOGGER.info('Starting discover')
-    check_accounts_list(config)
+    check_accounts_list(config) # Check that accounts are valid numbers.
+    client.check_accounts(config) # Check that accounts are valid LinkedIn Ads account.
```
Comment (karanpanchal-crest) [inline tap_linkedin_ads/__init__.py:25]: @savan-chovatiya There should be a single function for checking the accounts list. Can you make changes to the check_accounts_list() function to also check if the account is valid or not?

File: tap_linkedin_ads/client.py [hunk 2]
```diff
@@ -130,6 +130,33 @@ def check_access_token(self): #pylint: disable=inconsistent-return-statements
...
+                response = self.__session.get(
+                    url='https://api.linkedin.com/v2/adAccountUsersV2?q=accounts&count=1&start=0&accounts=urn:li:sponsoredAccount:{}'.format(account),
+                    headers=headers)
+                # Account users API will return 404 if provided account is valid number but invalid LinkedIn Ads account
+                if response.status_code == 404:
+                    invalid_account.append(account)
+                elif response.status_code != 200:
+                    raise_for_error(response)
+            if invalid_account:
+                error_message = 'These accounts are invalid LinkedIn Ads accounts from provided accounts:{}'.format(invalid_account)
```
Comment (karanpanchal-crest) [inline tap_linkedin_ads/client.py:157]: @savan-chovatiya The error message should be: "Invalid Linked Ads accounts provided during the configuration: [a1, a2. a3]"

File: tap_linkedin_ads/discover.py [hunk 3]
```diff
@@ -18,13 +18,3 @@ def discover():
...
-
-def check_accounts_list(config):
-    if config.get('accounts'):
-        account_list = config['accounts'].replace(" ", "").split(",")
-        for account in account_list:
-            try:
-                int(account)
-            except ValueError as e:
-                message = "The account '{}' provided in the configuration is having non-numeric value.".format(account)
-                raise ValueError(message) from None
```
Comment (savan-chovatiya) [inline tap_linkedin_ads/discover.py:30]: Removed this function as we have added API calls for validating accounts. Those calls will return 400 code for accounts with an invalid number format.

File: tap_linkedin_ads/__init__.py (hunk 1 above)
Comment (savan-chovatiya) [reply tap_linkedin_ads/__init__.py]: @karanpanchal-crest I have removed check_accounts_list() function and handled that scenario in check_accounts() function itself.

File: tap_linkedin_ads/client.py (hunk 2 above)
Comment (savan-chovatiya) [reply tap_linkedin_ads/client.py]: @karanpanchal-crest Updated error message as per suggestion.

**Review by dbshah1212 — APPROVED** [review]

**Review by KAllan357 — CHANGES_REQUESTED** [review]
Please fix conflicts

**Review by KAllan357 — APPROVED** [review]

**Pull request merged**